Entities/Structures/Common/AddTilesBySector.as

```diff
@@ -20,3 +20,28 @@ void AddTilesBySector(Vec2f ul, Vec2f lr, string sectorName, TileType tile, Tile
 		}
 	}
 }
+
+void AddTilesBySectorWithWindow(Vec2f ul, Vec2f lr, string sectorName, TileType tile, TileType omitTile = 255)
+{
+	if (getNet().isServer())
+	{
+		CMap@ map = getMap();
+		const f32 tilesize = map.tilesize;
+		Vec2f tpos = ul;
+		Vec2f tsize(map.tilesize, map.tilesize);
+		Vec2f center = ul + (lr - ul - tsize) * 0.5f;
+		while (tpos.x < lr.x)
+		{
+			while (tpos.y < lr.y)
+			{
+				if (map.getSectorAtPosition(tpos, sectorName) !is null && map.getTile(tpos).type != omitTile && tpos != center)
```

Comment:
Note: exact floating-point comparisons can cause issues due to precision.  
I have checked in engine, though, and `Vec2f::opCmp` luckily uses an epsilon value, so it is not an exact comparison. It seems to be OK in this usecase.